### PR TITLE
feat: optional/lazy imports

### PR DESCRIPTION
**Pull request by agucova**
This removes `pandas` and `matplotlib` as required dependencies, but lazily enables their related features when the modules are available, cutting import time significantly (> 2x). It also adds these as [extras](https://peps.python.org/pep-0508/#extras), installable with `pip install squigglepy[plots]` (for plotting-related functionality, matplotlib for now), `pip install squigglepy[ecosystem]` (for pandas, and in the future other related packages), or `pip install squigglepy[all]` (for all extras).

### CODE REVIEW

**Comment by peterhurford** [comment]
Nice! I'm seeing 448.81ms load time, down from 543.2ms (17% reduction)

File: squigglepy/distributions.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import numpy as np
-from matplotlib import pyplot as plt
 from scipy import stats
 
-from .utils import _process_weights_values, _is_numpy, is_dist, _round
+from .utils import _process_weights_values, _is_numpy, is_dist, _round, _optional_import
 from .version import __version__
 
+# We only import matplotlib.pyplot if we need it
+plt = _optional_import("matplotlib.pyplot")
```
Comment (peterhurford) [inline squigglepy/distributions.py:10]: Do you want to only evaluate this within the plot function?

**Comment by agucova** [comment]
@peterhurford how are you measuring load time? On my M1 this is at least a 2x improvement (this includes Python setup type):

image

image

**Comment by peterhurford** [comment]
@agucova measuring load time via https://github.com/rethinkpriorities/squigglepy/blob/main/tests/integration.py#L184-L190

File: squigglepy/distributions.py (hunk 1 above)
Comment (agucova) [reply squigglepy/distributions.py:10]: It's not necessary; the module only gets imported when it's actually used.

Comment (peterhurford) [reply squigglepy/distributions.py:10]: nice

**Pull request merged**